cmd/runtimetest/main.go

```diff
@@ -241,6 +242,67 @@ func validateROPaths(spec *rspec.Spec) error {
 			return fmt.Errorf("%v should be readonly", v)
 		}
 	}
+
+	return nil
+}
+
+func mountMatch(specMount rspec.Mount, sysMount rspec.Mount) error {
+	if specMount.Destination != sysMount.Destination {
+		return fmt.Errorf("mount destination expected: %v, actual: %v", specMount.Destination, sysMount.Destination)
+	}
+
+	if specMount.Type != sysMount.Type {
+		return fmt.Errorf("mount %v type expected: %v, actual: %v", specMount.Destination, specMount.Type, sysMount.Type)
+	}
+
+	if specMount.Source != sysMount.Source {
+		return fmt.Errorf("mount %v source expected: %v, actual: %v", specMount.Destination, specMount.Source, sysMount.Source)
+	}
+
+	optMap := make(map[string]bool)
+	for _, opt := range sysMount.Options {
+		optMap[opt] = true
+	}
+	for _, option := range specMount.Options {
+		if _, ok := optMap[option]; !ok {
+			return fmt.Errorf("mount %v option %v does not exist", specMount.Destination, option)
+		}
+	}
+
+	return nil
+}
+
+func validateMountsExistence(spec *rspec.Spec) error {
+	fmt.Println("validating mounts existence")
+	infos, err := mount.GetMounts()
+	if err != nil {
+		return err
+	}
+
+	mountsMap := make(map[string][]rspec.Mount)
+	for _, info := range infos {
+		m := rspec.Mount{
+			Destination: info.Mountpoint,
+			Type:        info.Fstype,
+			Source:      info.Source,
+			Options:     append(strings.Split(info.Opts, ","), strings.Split(info.VfsOpts, ",")...),
+		}
+		mountsMap[info.Mountpoint] = append(mountsMap[info.Mountpoint], m)
+	}
+
+	for _, specMount := range spec.Mounts {
+		found := false
+		for _, sysMount := range mountsMap[specMount.Destination] {
```

Comment:
Different mounts may have the same 'destination', but using 'destination' as a key in mountsMap is still an efficient way. So I use an array (`[]rspec.Mount`) as the value of mountsMap.